dapp/src/components/buySell/SwapCurrencyPill.js

```diff
@@ -418,16 +429,24 @@ const SwapCurrencyPill = ({
             {topItem && (
               <input
                 type="text"
-                value={coinValue}
+                value={formatMax(coinValue)}
                 placeholder="0.00"
                 onChange={(e) => {
+                  console.log('change')
                   const value = truncateDecimals(e.target.value)
                   const valueNoCommas = removeCommas(value)
                   if (checkValidInputForCoin(valueNoCommas, selectedCoin)) {
+                    if (maxBalanceSet && e.target.value === value)
+                      setMaxBalanceSet(false)
                     onAmountChange(valueNoCommas)
                   }
                 }}
                 onBlur={(e) => {
+                  setMaxBalanceSet(
+                    displayBalance &&
+                      Number(displayBalance.detailedBalance) ===
+                        Number(coinValue)
+                  )
                   const valueRounded = removeCommas(
```

Comment:
Also `onBlur` still seems to set the value of the swap calling `onAmountChange` with the rounded amount. So by: 
- clicking on `Max` button
- focusing into the input field
- focusing out (by clicking `Swap` lets say)
- the `onBlur` will be called, by rounding the amount and leaving the `dust` in the user's wallet

Reply:
fixed